api/v1alpha1/dns_types.go

```diff
@@ -7,6 +7,27 @@ package v1alpha1
 
 import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+// DNSLookupFamily defines the behavior of Envoy when resolving DNS for hostnames
+// +enum
+// +kubebuilder:validation:Enum=IPv4Only;IPv6Only;IPv4Preferred;IPv6Preferred;IPv4AndIPv6
+type DNSLookupFamily string
+
+const (
+	// IPv4Only means the DNS resolver will first perform a lookup for addresses in the IPv4 family.
```

Comment:
thoughts on just `IPv4` ? its less to type, but will it confuse the user who is aware of the envoyproxy api enums?